arbor/memory/cuda_wrappers.cpp

```diff
@@ -71,6 +71,10 @@ void cuda_free(void* ptr) {
     }
 }
 
+void cuda_sync() {
```

Comment:
This isn't needed (see comment where it is called)

Reply:
Removed based on discussion that basically concluded with the observation that `cudaDeviceSynchronize()` is only called when dealing with managed memory code. With total removal of managed memory, there should be no need for such a function wrapper.